src/pudl/transform/phmsagas.py

```diff
@@ -0,0 +1,164 @@
+        "excavation_damage_other",
+        "excavation_damage_total",
+        "excavation_tickets",
+        "services_efv_in_system",
+        "services_efv_installed",
+        "services_shutoff_valve_in_system",
+        "services_shutoff_valve_installed",
+        "federal_land_leaks_repaired_or_scheduled",
+    ],
+    "capitalization_exclusion": ["headquarters_address_state", "office_address_state"],
+    "cols_for_state_standardization": [
+        "headquarters_address_state",
+        "office_address_state",
+    ],
+}
+
+##############################################################################
+# PHMSAGAS transformation logic
+##############################################################################
+
+
+@asset(
+    ins={"raw_data": AssetIn(key="raw_phmsagas__yearly_distribution")},
+    io_manager_key=None,  # TODO: check on this one ("pudl_io_manager"? something else?)
+    compute_kind=None,  # TODO: check on this one
+)
+def core_phmsagas__yearly_distribution_operators(
+    raw_data: pd.DataFrame,
+) -> pd.DataFrame:
+    """Build the :ref:`core_phmsagas__yearly_distribution_operators`."""
+    df = raw_data[YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_to_keep"]]
+    df[YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_to_convert_to_ints"]] = df[
+        YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_to_convert_to_ints"]
+    ].astype("Int64")
+
+    # Ensure all "report_year" values have four digits
+    mask = df["report_year"] < 100
+
+    # Convert 2-digit years to appropriate 4-digit format (assume cutoff at year 50)
+    # We could also use the first 4 digits of the "report_number" but there was at least one anomaly here with an invalid year
+    df.loc[mask, "report_year"] = df.loc[mask, "report_year"].apply(
+        lambda x: 2000 + x if x < 50 else 1900 + x
+    )
+
+    # Fill NA values with zeroes because these columns are simply counts.
+    # Note that "excavation_damage..." columns should sum up to the value in "excavation_damage_total". However, many rows
+    # (on the scale of thousands) do not actually sum up to "excavation_damage_total".
+    df[YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_with_nas_to_fill"]] = df[
+        YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["columns_with_nas_to_fill"]
+    ].fillna(0)
+
+    # Fill in bad strings
+    df = fix_na(df)
+
+    # Standardize case for city, county, operator name, etc.
+    # Capitalize the first letter of each word in all object-type columns except the excluded ones
+    df[
+        df.select_dtypes(include=["object"]).columns.difference(
+            YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["capitalization_exclusion"]
+        )
+    ] = df[
+        df.select_dtypes(include=["object"]).columns.difference(
+            YEARLY_DISTRIBUTION_OPERATORS_COLUMNS["capitalization_exclusion"]
+        )
+    ].apply(lambda col: col.str.title())
+
+    # List of state columns to standardize
+    df = standardize_state_columns(
```

Comment:
I think we've used an `enum` constraint for the state column in EPA CEMS because it's a huge table and we want it to be interpreted as a categorical value to save space, and we don't use foreign key constraints in the hourly data that's only written to parquet. But for tables going into the database I think we would constrain the state values with a foreign key constraint that points at the `core_pudl__codes_subdivisions` table, and use the encoder setup to do the cleaning.

Reply:
Getting rid of my duplicate function and utilizing `POLITICAL_SUBDIVISIONS`. Plus adding an `enum` constraint to "state" now.

Reply:
Updated